src/runtime-rs/crates/runtimes/virt_container/src/sandbox.rs

```diff
@@ -223,6 +227,43 @@ impl VirtSandbox {
         })
     }
 
+    async fn prepare_vm_socket_config(&self) -> Result<ResourceConfig> {
```

Comment:
Similar comment as suggested in your initial proposal : please introduce this function with no functional change in a preliminary commit and only bring support for legacy vsock in a second commit. Separating refactoring changes from functional changes is a good practice that makes review and maintenance a lot easier.